### ISSUE TITLE
[Bug]: Detected blocking call to import_module inside the event loop by custom integration 'waste_collection_schedule' 

### ISSUE THREAD

**Issue by TheZoker** [mampfes/hacs_waste_collection_schedule#2121]
### I Have A Problem With:

The integration in general

### What's Your Problem

Since an update to version 2024.6 (beta) I get this warning in the logs:

```log
Detected blocking call to import_module inside the event loop by custom integration 'waste_collection_schedule' at 
custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py, line 188: source_module = 
importlib.import_module( (offender: /config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py, line 188: 
source_module = importlib.import_module(), please report it to the author of the 'waste_collection_schedule' custom 
integration
```

I think @bdraco could help fix this, since he is hunting all the blocking I/O calls and tried to eliminate them. 

### Checklist Source Error

- [ ] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [ ] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
Duplicate of #2023

**Comment by 5ila5** [comment]
Thought this was a duplicate of #2023 and it was resolved after the merge of @larena1 pr (#2041) (https://github.com/mampfes/hacs_waste_collection_schedule/issues/2023#issuecomment-2088960121)

But `source_module = importlib.import_module` seems to still cause this warning

**Comment by 5ila5** [comment]
I don't see this message with the dev container of https://github.com/home-assistant/core for everything after #2041 I only get this for versions before this PR. Do you use the master version or an older one?

**Comment by TheZoker** [comment]
I just checked and if I apply the changes from that PR, the issue seems to be gone. I have not updated yet, since there was no update for this integration available. This issue can be closed then
